### PR TITLE
ClickStreamFlow Update

### PR DESCRIPTION
**Pull request by achalpandey**
Accommodating changes suggested after merge in PR #8

### CODE REVIEW

File: ClickStreamFlow/README.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+ To run the app in the Standalone Runtime Environment:
+
+     $ ./run_standalone.sh /path/to/ClickStreamFlow-<version>.jar co.cask.tigon.apps.clickStreamflow.ClickStreamFlow
+
+ To run the app in the Distributed Runtime Environment:
+
+     $ ./run_distributed.sh /path/to/ClickStreamFlow-<version>.jar co.cask.tigon.apps.clickstreamflow.ClickStreamFlow
+
+Optional runtime arguments:
+    - ``httpPort``: Specify the netowrk port for the HTTP data ingestion service. If not specified, Tigon automatically assigns a port. This value is always announced at runtime.
```
Comment (awholegunch) [inline ClickStreamFlow/README.md]: This didn't get fixed: netowrk -> network

File: ClickStreamFlow/README.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    - ``tcpPort_viewStream``: Specify the network for the TCP data ingestion service for ingesting view stream data. If not specified, Tigon automatically assigns a port. This value is always announced at runtime.
```
Comment (awholegunch) [inline ClickStreamFlow/README.md]: Specify the network port for... would that not be more accurate?

File: ClickStreamFlow/README.md [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    - ``tcpPort_clickStream``: Specify the network for the TCP data ingestion service for ingesting click stream data. If not specified, Tigon automatically assigns a port. This value is always announced at runtime.
```
Comment (awholegunch) [inline ClickStreamFlow/README.md]: Specify the network port for... would that not be more accurate?

File: ClickStreamFlow/README.md [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    - {"data":["2","20","0","1","2","PageName2","'0':'LinkName0' - '1':'LinkName1' - '2':'LinkName2'"]}
+    - {"data":["3","30","0","1","2","PageName3","'0':'LinkName0' - '1':'LinkName1' - '2':'LinkName2'"]}
+    - {"data":["4","40","0","1","2","PageName4","'0':'LinkName0' - '1':'LinkName1' - '2':'LinkName2'"]}
+    - {"data":["5","50","0","1","2","PageName5","'0':'LinkName0' - '1':'LinkName1' - '2':'LinkName2'"]}
+    - {"data":["6","60","0","1","2","PageName6","'0':'LinkName0' - '1':'LinkName1' - '2':'LinkName2'"]}
+    - {"data":["7","70","0","1","2","PageName7","'0':'LinkName0' - '1':'LinkName1' - '2':'LinkName2'"]}
+    - {"data":["8","80","0","1","2","PageName8","'0':'LinkName0' - '1':'LinkName1' - '2':'LinkName2'"]}
+    - {"data":["9","90","0","1","2","PageName9","'0':'LinkName0' - '1':'LinkName1' - '2':'LinkName2'"]}
+    - {"data":["10","100","0","1","2","PageName10","'0':'LinkName0' - '1':'LinkName1' - '2':'LinkName2'"]}
+- ``clickStream`` :
```
Comment (awholegunch) [inline ClickStreamFlow/README.md]: Insert blank line before "clickStream"

File: ClickStreamFlow/README.md [hunk 5]
```diff
@@ -0,0 +1,111 @@
...
+    - pageInfo : This field contains additional information about the web page being viewed. It may contain the name of the page or the web server it is being served from.
+    - linkDetails : This field is a string-encoded map of link IDs and their associated details.
+- ``clickStream`` : A stream of data packets that represents a click event. It contains these fields:
+    - refPageViewID : The corresponding page view event ID that was responsible for the click.
+    - clickTime : The time at which this click event occured.
+    - lid : The ID of the link that was clicked. 
+
+The SQL query performs a join on these two data streams based on the pageViewID and the refPageViewID in viewStream and clickStream respectively. It generates an output record containing meta information associated with each click: click time; referrer page information; link ID; link details; and the referrer page view ID.
+
+The query filters out all click events that occurred five minutes (300 seconds) after the associated view event
```
Comment (awholegunch) [inline ClickStreamFlow/README.md]: occur five minutes (300 seconds) after the associated view event, [occur, add trailing period]

File: ClickStreamFlow/README.md [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+
+     mvn clean package
+
+ To build the jar without running the tests, run:
+
+     mvn clean package -DskipTests
+
+ To run the app in the Standalone Runtime Environment:
+
+     $ ./run_standalone.sh /path/to/ClickStreamFlow-<version>.jar co.cask.tigon.apps.clickStreamflow.ClickStreamFlow
```
Comment (awholegunch) [inline ClickStreamFlow/README.md]: co.cask.tigon.apps.clickstreamflow.ClickStreamFlow

File: ClickStreamFlow/README.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+This app serves as a demonstration for using Tigon SQL to join 2 data streams.
+
+In this example, a view event data stream and a click event data stream are joined to generate meta information for each click event using the view event data stream. Click events are filtered on the basis of conditions specified in a SQL query.
+
+## Implementation Details
+
+The ``ClickStreamFlow`` application consists of:
+ - A ``SQLInputFlowlet`` that accepts two incoming data streams of view events and click events. It then processes this data based on the provided SQL query and provides meta information for each click event using the information from the associated view event. 
+ - A ``DigestFlowlet`` that accepts the output data generated by the ``SQLInputFlowlet`` and emits it to a HTTP end-point as a string encoded JSON. 
+ - The generated output data is logged to stdout.
```
Comment (gokulavasan) [inline ClickStreamFlow/README.md:14]: The generated output data is _also_ logged to stdout?

Comment (achalpandey) [inline ClickStreamFlow/README.md:14]: We never referenced "logging to stdout" before this point. I don't think we need to add "also" to the above statement.

**Comment by awholegunch** [comment]
LGTM.

**Pull request merged**